### ISSUE TITLE
Docker for mac doesn't restart after upgrade to Version 1.12.3 (13776)

### ISSUE THREAD

**Issue by maximede** [docker/for-mac#954]
Hello, After upgrading to the latest version of Docker for Mac this morning, the application is not starting.

### Expected behavior
Docker for mac starts as usual

### Actual behavior

After launching the app, it stays on "Docker starting..."
Tried clicking restart and restarting the app, it did not help.

### Information

Docker for Mac: version: 1.12.3 (583d1b8)
OS X: version 10.12.1 (build: 16B2555)
logs: /tmp/FC561B12-DB93-43D6-8E55-C7AA5F0A3411/20161115-124033.tar.gz
failure: docker ps failed: (Failure "docker ps: timeout after 10.00s")
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     db
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     slirp
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         docker ps failed
[OK]     menubar
[OK]     disk
Failure: Could not upload diagnostic data to remote server (docker-diagnose exit code is 1)

From `syslog -k Sender Docker`

```
Nov 15 12:47:30 REDACTED Docker[8113] <Notice>: EOF on watchdog pipe: shutting down
Nov 15 12:47:30 REDACTED Docker[8113] <Notice>: Received SIGTERM, shutting down
Nov 15 12:47:30 REDACTED Docker[8113] <Notice>: Stop 1 children with order 1: com.docker.driver.amd64-linux (pid 8122)
Nov 15 12:47:30 REDACTED Docker[8113] <Notice>: Signal terminated to com.docker.driver.amd64-linux (pid 8122)
Nov 15 12:47:32 REDACTED Docker[8826] <Notice>: Client reports version 1, commit 583d1b8ffe2e3d117e4725a0857d335bc27190ac
Nov 15 12:47:32 REDACTED Docker[8826] <Notice>: EOF reading packet from Unix domain socket: closing
Nov 15 12:47:32 REDACTED Docker[8826] <Error>: Failed to read command from client
Nov 15 12:47:32 REDACTED Docker[8827] <Notice>: Client reports version 1, commit 583d1b8ffe2e3d117e4725a0857d335bc27190ac
Nov 15 12:47:32 REDACTED Docker[8827] <Critical>: client with uid 501 requests com.docker.vmnetd install symlinks
Nov 15 12:47:32 REDACTED Docker[8827] <Critical>: File /usr/local/bin/docker is a symlink
Nov 15 12:47:32 REDACTED Docker[8827] <Notice>: Client supplied container path /Users/REDACTED/Library/Containers/com.docker.docker/Data
Nov 15 12:47:32 REDACTED Docker[8827] <Critical>: File /var/run/docker.sock is a symlink or a socket
Nov 15 12:47:32 REDACTED Docker[8827] <Notice>: Symlinking /var/run/docker.sock to /Users/REDACTED/Library/Containers/com.docker.docker/Data/s60
Nov 15 12:47:32 REDACTED Docker[8827] <Critical>: File /usr/local/bin/docker-compose is a symlink
Nov 15 12:47:32 REDACTED Docker[8827] <Notice>: Client supplied container path /Users/REDACTED/Library/Containers/com.docker.docker/Data
Nov 15 12:47:32 REDACTED Docker[8827] <Critical>: File /var/run/docker.sock is a symlink or a socket
Nov 15 12:47:32 REDACTED Docker[8827] <Notice>: Symlinking /var/run/docker.sock to /Users/REDACTED/Library/Containers/com.docker.docker/Data/s60
Nov 15 12:47:32 REDACTED Docker[8827] <Critical>: File /usr/local/bin/notary is a symlink
Nov 15 12:47:32 REDACTED Docker[8827] <Notice>: Client supplied container path /Users/REDACTED/Library/Containers/com.docker.docker/Data
Nov 15 12:47:32 REDACTED Docker[8827] <Critical>: File /var/run/docker.sock is a symlink or a socket
Nov 15 12:47:32 REDACTED Docker[8827] <Notice>: Symlinking /var/run/docker.sock to /Users/REDACTED/Library/Containers/com.docker.docker/Data/s60
Nov 15 12:47:32 REDACTED Docker[8827] <Critical>: File /usr/local/bin/docker-machine is a symlink
Nov 15 12:47:32 REDACTED Docker[8827] <Notice>: Client supplied container path /Users/REDACTED/Library/Containers/com.docker.docker/Data
Nov 15 12:47:32 REDACTED Docker[8827] <Critical>: File /var/run/docker.sock is a symlink or a socket
Nov 15 12:47:32 REDACTED Docker[8827] <Notice>: Symlinking /var/run/docker.sock to /Users/REDACTED/Library/Containers/com.docker.docker/Data/s60
Nov 15 12:47:38 REDACTED Docker[8122] <Notice>: Docker is not responding: Get http://./info: EOF: waiting 0.5s
--- last message repeated 4 times ---
Nov 15 12:48:30 REDACTED Docker[8113] <Warning>: Timed-out waiting for graceful shutdown
Nov 15 12:48:30 REDACTED Docker[8113] <Warning>: Sending a SIGKILL to /Applications/Docker.app/Contents/MacOS/com.docker.driver.amd64-linux
Nov 15 12:48:30 REDACTED Docker[8113] <Notice>: Signal killed to com.docker.driver.amd64-linux (pid 8122)
Nov 15 12:48:30 REDACTED Docker[8113] <Notice>: Reap com.docker.driver.amd64-linux (pid 8122): signal: killed
Nov 15 12:48:30 REDACTED Docker[8113] <Notice>: Stop 2 children with order 2: com.docker.osxfs (pid 8119), com.docker.slirp (pid 8120)
Nov 15 12:48:30 REDACTED Docker[8113] <Notice>: Signal terminated to com.docker.osxfs (pid 8119)
Nov 15 12:48:30 REDACTED Docker[8113] <Notice>: Signal terminated to com.docker.slirp (pid 8120)
Nov 15 12:48:30 REDACTED Docker[8113] <Notice>: Reap com.docker.slirp (pid 8120): signal: terminated
Nov 15 12:48:30 REDACTED Docker[8113] <Notice>: Reap com.docker.osxfs (pid 8119): signal: terminated
Nov 15 12:48:30 REDACTED Docker[8113] <Notice>: Stop 1 children with order 3: com.docker.osx.hyperkit.linux (pid 8121)
Nov 15 12:48:30 REDACTED Docker[8113] <Notice>: Signal terminated to com.docker.osx.hyperkit.linux (pid 8121)
Nov 15 12:48:30 REDACTED Docker[8113] <Notice>: Reap com.docker.osx.hyperkit.linux (pid 8121): signal: terminated
Nov 15 12:48:30 REDACTED Docker[8113] <Notice>: Stop 1 children with order 4: com.docker.db (pid 8118)
Nov 15 12:48:30 REDACTED Docker[8113] <Notice>: Signal terminated to com.docker.db (pid 8118)
Nov 15 12:48:30 REDACTED Docker[8113] <Notice>: Reap com.docker.db (pid 8118): exit status 1
Nov 15 12:48:30 REDACTED Docker[8828] <Notice>: Acquired task manager lock
Nov 15 12:48:30 REDACTED Docker[8828] <Notice>: Maximum number of file descriptors is 10240
Nov 15 12:48:30 REDACTED Docker[8828] <Notice>: Removing leaked pidfile com.docker.hyperkit
Nov 15 12:48:30 REDACTED Docker[8828] <Notice>: Starting com.docker.db, com.docker.osxfs, com.docker.slirp, com.docker.osx.hyperkit.linux, com.docker.driver.amd64-linux
Nov 15 12:48:30 REDACTED Docker[8828] <Notice>: Start com.docker.db (pid 8861)
Nov 15 12:48:30 REDACTED Docker[8828] <Notice>: Start com.docker.osxfs (pid 8862)
Nov 15 12:48:30 REDACTED Docker[8828] <Notice>: Start com.docker.slirp (pid 8863)
Nov 15 12:48:30 REDACTED Docker[8828] <Notice>: Start com.docker.osx.hyperkit.linux (pid 8864)
Nov 15 12:48:30 REDACTED Docker[8828] <Notice>: Start com.docker.driver.amd64-linux (pid 8865)
Nov 15 12:48:30 REDACTED Docker[8864] <Notice>: API server starting
Nov 15 12:48:30 REDACTED Docker[8864] <Notice>: 🍀 socket path is: /Users/REDACTED/Library/Containers/com.docker.docker/Data/s20
Nov 15 12:48:30 REDACTED Docker[8864] <Notice>: API socket removed
Nov 15 12:48:30 REDACTED Docker[8862] <Notice>: Logging to Apple System Log
Nov 15 12:48:30 REDACTED Docker[8864] <Notice>: VM shutdown at 2016-11-15 12:38:07 -0800 PST
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: Acquired hypervisor lock
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: Docker is not responding: Get http://./info: dial unix /Users/REDACTED/Library/Containers/com.docker.docker/Data/*00000003.00000948: connect: connection refused: waiting 0.5s
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: Logging to Apple System Log
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: Setting handler to ignore all SIGPIPE signals
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: vpnkit version 7584d230f7d9457125dfa614cc13425e0b2e93be with hostnet version local  uwt version 0.0.3 hvsock version 0.10.0
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: starting port_forwarding port_control_url:fd:4 vsock_path:/Users/REDACTED/Library/Containers/com.docker.docker/Data/@connect
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: hosts file has bindings for broadcasthost localhost 
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: OSX version = 10.12.1, default value of on-sleep = no not freeze
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: hypervisor: native
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: filesystem: osxfs
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: network: hybrid
Nov 15 12:48:30 REDACTED Docker[8862] <Notice>: Using protocol TwoThousand msize 16384
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: Hypervisor: native; BootProtocol: direct; UefiBootDisk: /Users/REDACTED/UefiBoot.qcow2
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: Syslog socket is /Users/REDACTED/Library/Containers/com.docker.docker/Data/*00000002.00000202
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: Logfile is /Users/REDACTED/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/syslog
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: Launched[8870]: /Applications/Docker.app/Contents/MacOS/com.docker.hyperkit -A -m 2G -c 4 -u -s 0:0,hostbridge -s 31,lpc -s 2:0,virtio-vpnkit,uuid=943ebabe-8948-4435-a5c9-31ff679d971f,path=/Users/REDACTED/Library/Containers/com.docker.docker/Data/s50,macfile=/Users/REDACTED/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/mac.0 -s 3,virtio-blk,file:///Users/REDACTED/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.qcow2?sync=1&buffered=1,format=qcow -s 4,virtio-9p,path=/Users/REDACTED/Library/Containers/com.docker.docker/Data/s40,tag=db -s 5,virtio-rnd -s 6,virtio-9p,path=/Users/REDACTED/Library/Containers/com.docker.docker/Data/s51,tag=port -s 7,virtio-sock,guest_cid=3,path=/Users/REDACTED/Library/Containers/com.docker.docker/Data,guest_forwards=2376;1525 -l com1,autopty=/Users/REDACTED/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/tty,log=/Users/REDACTED/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/console-ring -f kexec,/Applications/Docker.app/Contents/Resources/moby/vmlinuz64,/Applications/Docker.app/Contents/Resources/moby/initrd.img,earlyprintk=serial console=ttyS0 com.docker.driver="com.docker.driver.amd64-linux", com.docker.database="com.docker.driver.amd64-linux" ntp=gateway mobyplatform=mac -F /Users/REDACTED/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/hypervisor.pid
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: SC database lists DNS servers: REDACTED, REDACTED
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: SC database lists search domains: REDACTED
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: attempting to reconnect to database
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: reconnected transport layer
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: updating connection limit to 900
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: updating search domains to REDACTED
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: updating resolvers to nameserver REDACTED#53
	nameserver REDACTED#53
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: using DNS forwarders on REDACTED#53; REDACTED#53
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: allowing binds to any IP addresses
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: Creating slirp server pcap_settings:disabled peer_ip:192.168.65.2 local_ip:192.168.65.1 domain_search:REDACTED
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: PPP.negotiate: received ((magic VMN3T)(version 1)(commit 583d1b8ffe2e3d117e4725a0857d335bc27190ac))
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: PPP.negotiate: received (Ethernet 943ebabe-8948-4435-a5c9-31ff679d971f)
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: PPP.negotiate: sending ((mtu 1500)(max_packet_size 1550)(client_macaddr c0:ff:ee:c0:ff:ee))
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: virtio-net-vpnkit: magic=VMN3T version=1 commit=0123456789012345678901234567890123456789
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: PPP.listen: called a second time: doing nothing
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: TCP/IP ready
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: stack connected
Nov 15 12:48:30 REDACTED Docker[8863] <Notice>: starting introspection intrspection_url:fd:5
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: mirage_block_open: file:///Users/REDACTED/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.qcow2?sync=1&buffered=1
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: mirage_block_open: file:///Users/REDACTED/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.qcow2?sync=1&buffered=1 returning 0
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: mirage_block_stat
Nov 15 12:48:30 REDACTED Docker[8865] <Notice>: vsock init 7:0 = /Users/REDACTED/Library/Containers/com.docker.docker/Data, guest_cid = 00000003
Nov 15 12:48:32 REDACTED Docker[8865] <Notice>:
	rdmsr to register 0x34 on vcpu 0
Nov 15 12:48:33 REDACTED Docker[8865] <Notice>: Docker is not responding: Get http://./info: EOF: waiting 0.5s
--- last message repeated 4 times ---
Nov 15 12:48:41 REDACTED Docker[8863] <Notice>: BOOTREQUEST from c0:ff:ee:c0:ff:ee
Nov 15 12:48:41 REDACTED Docker[8863] <Notice>: BOOTREPLY to c0:ff:ee:c0:ff:ee yiddr 192.168.65.2 siddr 192.168.65.1 dns 192.168.65.1, 192.168.65.10, 192.168.65.9, 192.168.65.8, 192.168.65.7, 192.168.65.6, 192.168.65.5, 192.168.65.4, 192.168.65.3 router 192.168.65.1 domain local
Nov 15 12:48:41 REDACTED Docker[8862] <Notice>: transfused: mount /bin/fusermount -o allow_other,max_read=1048576,subtype=osxfs /Users
Nov 15 12:48:41 REDACTED Docker[8862] <Notice>: transfused: mount /bin/fusermount -o allow_other,max_read=1048576,subtype=osxfs /Volumes
Nov 15 12:48:41 REDACTED Docker[8862] <Notice>: transfused: mount /bin/fusermount -o allow_other,max_read=1048576,subtype=osxfs /tmp
Nov 15 12:48:41 REDACTED Docker[8862] <Notice>: transfused: mount /bin/fusermount -o allow_other,max_read=1048576,subtype=osxfs /private
Nov 15 12:48:41 REDACTED Docker[8862] <Notice>: transfused: mount /bin/fusermount -o allow_other,max_read=1048576,subtype=osxfs /host_docker_app
Nov 15 12:48:41 REDACTED Docker[8862] <Notice>: Negotiated transfuse notification channel for /Users
Nov 15 12:48:41 REDACTED Docker[8862] <Notice>: Negotiated transfuse notification channel for /Volumes
Nov 15 12:48:41 REDACTED Docker[8862] <Notice>: Negotiated transfuse notification channel for /tmp
Nov 15 12:48:41 REDACTED Docker[8862] <Notice>: Negotiated transfuse notification channel for /private
Nov 15 12:48:41 REDACTED Docker[8862] <Notice>: Negotiated transfuse notification channel for /host_docker_app
Nov 15 12:48:41 REDACTED Docker[8862] <Notice>: sending continue to client
Nov 15 12:48:42 REDACTED Docker[8863] <Notice>: Using protocol TwoThousand msize 8192
Nov 15 12:48:44 REDACTED Docker[8865] <Notice>: Docker is not responding: Get http://./info: EOF: waiting 0.5s
```

Diagnostic ID : FC561B12-DB93-43D6-8E55-C7AA5F0A3411

### Steps to reproduce the behavior

  1. Upgrade to  1.12.3 (13776)

**Comment by hetenho** [comment]
I have the exact same problem.

**Comment by maximede** [comment]
Doing this would delete all the images and all the containers, right ?

**Comment by ijc25** [comment]
@maximede yes, a factory reset would delete all images and containers. However your diagnostic contains errors suggesting that the docker symlinks have not been correctly installed by the installer, which is a known issue that we are currently working on. A possible workaround is to install the docker client manually by running:
```
ln -s ~/Library/Group\ Containers/group.com.docker/bin/docker /usr/local/bin/docker
```

May be related to docker/for-mac#785

Please ing to reopen if the workaround does not help.

**Comment by maximede** [comment]
@ijc25 Just tried the command but the symlink was already present

```
lrwxr-xr-x  1 me  staff    73B Nov 17 16:36 /usr/local/bin/docker -> /Users/me/Library/Group Containers/group.com.docker/bin/docker
```

**Comment by ijc25** [comment]
`/var/log/docker.log` contains:

> Error creating cluster component: error while loading TLS Certificate in /var/lib/docker/swarm/certificates/swarm-node.crt: x509: certificate has expired or is not yet valid

and the docker daemon appears to have died/exited. I think this relates to https://github.com/docker/docker/issues/24132 but it's unclear how/why this certificate has expired. I'm fairly sure that a factory reset would "fix" this issue at the expense of throwing away your existing containers and pulled images.

It seems like you may have had some residual swarm configuration issues from when you were running 1.12.1. there are lots of messages like this dating from before you updated:
```
time="2016-11-10T23:52:54.166311997Z" level=error msg="agent: session failed" error="rpc error: code = 2 desc = grpc: the client connection is closing" module=agent 
time="2016-11-10T23:52:54.168164670Z" level=debug msg="agent: rebuild session" module=agent 
time="2016-11-10T23:52:54.201645652Z" level=debug msg="(*session).start" module=agent 
```

I think at this point my best advice would be to factory reset, sorry about that.

**Comment by maximede** [comment]
No worries, do you want me to run anything / send you any log before I do the factory reset ?

**Comment by ijc25** [comment]
@maximede

Entering a privileged shell with:
```
docker run -it --privileged --pid=host debian nsenter -t 1 -m -n /bin/sh
```

and running

```
openssl x509 -in /var/lib/docker/swarm/certificates/swarm-node.crt -text -noout
```

might be interesting, I'm curious what the expiry time time on that crt might be.

**Comment by maximede** [comment]
well, docker is not running :)

**Comment by ijc25** [comment]
Doh! `screen ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/tty` should get you a shell.

**Comment by maximede** [comment]
Here you go. Looks like it expired `Nov 10 23:30:00 2016 GMT`

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            4d:8d:cc:40:a3:3c:c2:62:1c:74:c5:90:6b:00:72:b8:1a:a2:4f:da
    Signature Algorithm: ecdsa-with-SHA384
        Issuer: CN=swarm-ca
        Validity
            Not Before: Aug 12 23:30:00 2016 GMT
            Not After : Nov 10 23:30:00 2016 GMT
        Subject: O=e35utc214iur8ibbm6a0egvee, OU=swarm-manager, CN=3o26jujg4jnf1ifbgbl28t4x5
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (256 bit)
                pub:
                    04:94:23:7d:02:2d:49:c9:0f:7b:f8:a8:a1:6a:79:
                    79:fd:6b:47:77:08:32:09:5e:b8:7e:1e:0d:75:65:
                    bd:06:00:3e:bc:10:12:33:ba:7d:e8:f8:c8:06:ce:
                    cb:62:83:c5:ff:61:41:71:c4:b6:bd:de:0c:ca:6d:
                    cb:a0:0a:9c:67
                ASN1 OID: prime256v1
                NIST CURVE: P-256
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage:
                TLS Web Server Authentication, TLS Web Client Authentication
            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Subject Key Identifier:
                AF:4E:A6:2E:E2:3B:8F:FB:BD:1B:74:03:07:BC:7E:1F:62:50:22:BF
            X509v3 Authority Key Identifier:
                keyid:4D:C4:83:C4:9A:A7:62:44:27:4B:62:FB:71:68:FF:33:4D:1F:CB:C6

            X509v3 Subject Alternative Name:
                DNS:swarm-manager, DNS:swarm-ca
    Signature Algorithm: ecdsa-with-SHA384
         30:66:02:31:00:b7:9c:2f:5b:9f:1b:fa:43:7f:b2:6f:5e:e2:
         33:bb:ba:eb:31:9c:b9:2b:2f:df:15:3a:60:bf:5a:cf:5f:9e:
         ab:d9:f8:19:f2:f8:e7:b5:d3:c3:39:3e:04:62:60:48:fe:02:
         31:00:b9:1c:3b:a6:06:3e:78:f8:56:83:ca:08:ef:9f:89:d0:
         68:ce:83:ae:3c:9f:41:e0:96:fa:10:30:58:e4:88:59:ec:1d:
         dd:2b:fe:81:3b:db:4a:d8:7e:42:5f:20:81:4c
```

**Comment by ijc25** [comment]
Thanks, that does correlate rather suspiciously closely with those grpc messages starting to show up in your logs about 20 mins later.

Do you remember how/when you initialised swarm mode? Sadly the diagnostics bundle does not collect much useful swarm mode info (I coincidentally improved that this week so the next betas should be better in this regard).

**Comment by maximede** [comment]
looks like it was in August 

`2016-08-12 16:35  docker swarm init`

**Comment by samoht** [comment]
The issue has been fixed in 1.13 (by https://github.com/docker/docker/pull/28228), so closing that issue. Thanks for the report!
